### ISSUE TITLE
Improves sendCookie handling

### ISSUE THREAD

**Issue by insraq** [CRAlpha/react-native-wkwebview#50]
1. sendCookie should actually synchronize the cookies from NSHTTPCookieStorage to document.cookie. This will enable AJAX requests to share the cookie. You can simply use document.cookie = newCookie; to set/update cookies.
2. Add a new prop like updateCookie to force an update from Set-Cookie header to NSHTTPCookieStorage. I believe Set-Cookie will eventually end up in NSHTTPCookieStorage, it's just a matter of when.
3. Maybe also expose webView:decidePolicyForNavigationResponse:decisionHandler: and WKUserScript.

**Comment by jacquescrocker** [comment]
Any thoughts on this? https://github.com/CRAlpha/react-native-wkwebview/pull/67

I'm using WkWebView to reuse my website's authentication system (fairly complex and cant easily be extracted into an api / oauth flow). So I'll need to watch and pull out a cookie from the environment once the user is logged in.

**Comment by wireforce** [comment]
Maybe instead of patching the current WKWebView we can implement proper cookie-handling in this component, when iOS11 is out. With iOS11 Apple seems to have added the proper tools to handle cookies:

https://developer.apple.com/documentation/webkit/wkhttpcookiestore
(info video: https://developer.apple.com/videos/play/wwdc2017/220/)

**Comment by insraq** [comment]
@wireforce I agree we should implement `WKHTTPCookieStore`. Feel free to make a PR if you are interested..
